> Unfortunately, calling the context menu raises an exception. Possible Python version problem?
> 
> ```
> Traceback (most recent call last):
>   File "flask\app.py", line 2073, in wsgi_app
>   File "flask\app.py", line 1518, in full_dispatch_request
>   File "flask\app.py", line 1516, in full_dispatch_request
>   File "flask\app.py", line 1502, in dispatch_request
>   File "isisInternal.py", line 28, in getNexusFields
>   File "nexusInteraction.py", line 88, in runFields
>   File "nexusInteraction.py", line 34, in dataFields
> TypeError: 'set' object is not subscriptable
> ```

Very strange, works fine on linux, must be a version error in the build like you suggested

--- a/.azure-pipelines/templates/windowsBuild.yml
+++ b/.azure-pipelines/templates/windowsBuild.yml
@@ -13,7 +13,7 @@ parameters:
 steps:
   - task: UsePythonVersion@0
     inputs:
-      versionSpec: '3.9.x'
+      versionSpec: '3.9.8'
   - script: |
       python3 -m pip install pyinstaller
       python3 -m pip install flask

--- a/.azure-pipelines/templates/windowsBuild.yml
+++ b/.azure-pipelines/templates/windowsBuild.yml
@@ -13,7 +13,7 @@ parameters:
 steps:
   - task: UsePythonVersion@0
     inputs:
-      versionSpec: '3.x'
+      versionSpec: '3.9'
   - script: |
       python3 -m pip install pyinstaller
       python3 -m pip install flask

--- a/.azure-pipelines/templates/windowsBuild.yml
+++ b/.azure-pipelines/templates/windowsBuild.yml
@@ -13,7 +13,7 @@ parameters:
 steps:
   - task: UsePythonVersion@0
     inputs:
-      versionSpec: '3.9.8'
+      versionSpec: '3.x'
   - script: |
       python3 -m pip install pyinstaller
       python3 -m pip install flask

--- a/backend/nexusInteraction.py
+++ b/backend/nexusInteraction.py
@@ -16,7 +16,7 @@ def file(instrument, cycle, run):
         nxsFile = File(nxsDir)
         return nxsFile
     except(Exception):
-        return {"response:", "ERR. File failed to open"}
+        return ["response:", "ERR. File failed to open"]
 
 # Get run times
 
@@ -70,7 +70,7 @@ def runData(file, fields, run):
                     value = dataBlock['value']
                     time = dataBlock['time']
                 except(Exception):
-                    return {"response:", "ERR. Data not found"}
+                    return ["response:", "ERR. Data not found"]
 
         blockShape = value.shape
         blockData.append([run, field])

